commandline/src/main/java/tech/pegasys/web3signer/commandline/subcommands/Eth2SubCommand.java

```diff
@@ -40,21 +44,35 @@
 
   public static final String COMMAND_NAME = "eth2";
 
-  @Spec CommandSpec spec;
+  @Spec CommandSpec commandSpec;
+
+  @CommandLine.Option(
+      names = {"-n", "--network"},
+      paramLabel = "<NETWORK>",
+      description = "Represents which network to use.",
+      arity = "1")
+  private String network = "mainnet";
 
   @Mixin private PicoCliSlashingProtectionParameters slashingProtectionParameters;
   @Mixin private PicoCliAzureKeyVaultParameters azureKeyVaultParameters;
+  private tech.pegasys.teku.spec.Spec eth2Spec;
 
   @Override
   public Runner createRunner() {
-    return new Eth2Runner(config, slashingProtectionParameters, azureKeyVaultParameters);
+    return new Eth2Runner(config, slashingProtectionParameters, azureKeyVaultParameters, eth2Spec);
   }
 
   @Override
   protected void validateArgs() {
+    final String networkConfigName =
+        Eth2Network.fromStringLenient(network).map(Eth2Network::configName).orElse(network);
```

Comment:
Won't this change use mainnet if an invalid network is provided? That seems a bit dangerous to me, rather see with fail.

Reply:
no, this will first attempt to use config name as defined in Eth2Network. If not available, then network will be used whatever user has specified (for example a URL or directory location). Teku uses `_` in few of the enums in Eth2Network. The fromStringLenient converts `-` to `_` to get the correct value. Copied verbatim from Teku.